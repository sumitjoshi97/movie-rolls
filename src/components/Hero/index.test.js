import React from 'react'
import { shallow } from 'enzyme'

import { Hero } from './index'

describe('Hero component unit test', () => {
  const addToList = jest.fn()
  const removeFromList = jest.fn()
  const setRedirectPath = jest.fn()

  const type = 'movie'
  const pageId = '0'
  const title = 'some'
  const poster = null
  const backdrop = null
  const rating = 4
  const year = '2019'
  const userId = '012'
  const favorite = {}
  const watch = {}

  const props = {
    type,
    pageId,
    title,
    poster,
    backdrop,
    rating,
    year,
    userId,
    favorite,
    watch,
    addToList,
    removeFromList,
    setRedirectPath
  }

  const heroWrapper = shallow(<Hero {...props} />)

  test('will render properly', () => {
    expect(heroWrapper).toMatchSnapshot()
  })

  test('clicking favorite btn calls addToList fn for favorite list', () => {
    const listData = {
      pageId,
      title,
      type,
      poster
    }

    heroWrapper
      .find('.hero__info__text__action-btns__favorite')
      .simulate('click')
    expect(addToList).toHaveBeenCalledWith('favorite', listData, userId)
  })

  test('clicking watch-later btn calls addToList fn for watchlist', () => {
    const listData = {
      pageId,
      title,
      type,
      poster
    }

    heroWrapper
      .find('.hero__info__text__action-btns__watch-later')
      .simulate('click')
    expect(addToList).toHaveBeenCalledWith('watch', listData, userId)
  })
})
