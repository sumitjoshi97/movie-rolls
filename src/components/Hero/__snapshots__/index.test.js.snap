// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component unit test will render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero
    addToList={[MockFunction]}
    backdrop={null}
    favorite={Object {}}
    pageId="0"
    poster={null}
    rating={4}
    removeFromList={[MockFunction]}
    setRedirectPath={[MockFunction]}
    title="some"
    type="movie"
    userId="012"
    watch={Object {}}
    year="2019"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="hero__info"
      >
        <Link
          onClick={[Function]}
          replace={false}
          to="/details/0"
        >
          <img
            alt=""
            className="hero__info__img"
            src="https://image.tmdb.org/t/p/w185/null"
          />
        </Link>
        <div
          className="hero__info__text"
        >
          <Link
            onClick={[Function]}
            replace={false}
            to="/details/0"
          >
            <h1
              className="hero__info__text__header"
            >
              some
            </h1>
            <div
              className="hero__info__text__rating"
            >
              4
              <RatingStars
                stars={4}
              />
            </div>
            <div
              className="hero__info__text__type"
            >
              <span>
                movie
              </span>
              
              <span
                className="hero__info__text__year"
              >
                2019
              </span>
            </div>
            <p
              className="hero__info__text__summary"
            />
          </Link>
          <div
            className="hero__info__text__action-btns"
          >
            <button
              className="hero__info__text__action-btns__favorite"
              onClick={[Function]}
            >
              add to 
               favorite
            </button>
            <button
              className="hero__info__text__action-btns__watch-later"
              onClick={[Function]}
            >
              add to 
               Watch Later
            </button>
          </div>
        </div>
      </div>,
      "className": "hero",
      "style": Object {
        "backgroundImage": "linear-gradient(to bottom, rgba(0,0,0,0.6), rgba(0,0,0,0.9)), url(https://image.tmdb.org/t/p/original/null)",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            onClick={[Function]}
            replace={false}
            to="/details/0"
          >
            <img
              alt=""
              className="hero__info__img"
              src="https://image.tmdb.org/t/p/w185/null"
            />
          </Link>,
          <div
            className="hero__info__text"
          >
            <Link
              onClick={[Function]}
              replace={false}
              to="/details/0"
            >
              <h1
                className="hero__info__text__header"
              >
                some
              </h1>
              <div
                className="hero__info__text__rating"
              >
                4
                <RatingStars
                  stars={4}
                />
              </div>
              <div
                className="hero__info__text__type"
              >
                <span>
                  movie
                </span>
                
                <span
                  className="hero__info__text__year"
                >
                  2019
                </span>
              </div>
              <p
                className="hero__info__text__summary"
              />
            </Link>
            <div
              className="hero__info__text__action-btns"
            >
              <button
                className="hero__info__text__action-btns__favorite"
                onClick={[Function]}
              >
                add to 
                 favorite
              </button>
              <button
                className="hero__info__text__action-btns__watch-later"
                onClick={[Function]}
              >
                add to 
                 Watch Later
              </button>
            </div>
          </div>,
        ],
        "className": "hero__info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt=""
              className="hero__info__img"
              src="https://image.tmdb.org/t/p/w185/null"
            />,
            "onClick": [Function],
            "replace": false,
            "to": "/details/0",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "hero__info__img",
              "src": "https://image.tmdb.org/t/p/w185/null",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                onClick={[Function]}
                replace={false}
                to="/details/0"
              >
                <h1
                  className="hero__info__text__header"
                >
                  some
                </h1>
                <div
                  className="hero__info__text__rating"
                >
                  4
                  <RatingStars
                    stars={4}
                  />
                </div>
                <div
                  className="hero__info__text__type"
                >
                  <span>
                    movie
                  </span>
                  
                  <span
                    className="hero__info__text__year"
                  >
                    2019
                  </span>
                </div>
                <p
                  className="hero__info__text__summary"
                />
              </Link>,
              <div
                className="hero__info__text__action-btns"
              >
                <button
                  className="hero__info__text__action-btns__favorite"
                  onClick={[Function]}
                >
                  add to 
                   favorite
                </button>
                <button
                  className="hero__info__text__action-btns__watch-later"
                  onClick={[Function]}
                >
                  add to 
                   Watch Later
                </button>
              </div>,
            ],
            "className": "hero__info__text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <h1
                    className="hero__info__text__header"
                  >
                    some
                  </h1>,
                  <div
                    className="hero__info__text__rating"
                  >
                    4
                    <RatingStars
                      stars={4}
                    />
                  </div>,
                  <div
                    className="hero__info__text__type"
                  >
                    <span>
                      movie
                    </span>
                    
                    <span
                      className="hero__info__text__year"
                    >
                      2019
                    </span>
                  </div>,
                  <p
                    className="hero__info__text__summary"
                  />,
                ],
                "onClick": [Function],
                "replace": false,
                "to": "/details/0",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "some",
                    "className": "hero__info__text__header",
                  },
                  "ref": null,
                  "rendered": "some",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      4,
                      <RatingStars
                        stars={4}
                      />,
                    ],
                    "className": "hero__info__text__rating",
                  },
                  "ref": null,
                  "rendered": Array [
                    4,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "stars": 4,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        movie
                      </span>,
                      "",
                      <span
                        className="hero__info__text__year"
                      >
                        2019
                      </span>,
                    ],
                    "className": "hero__info__text__type",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "movie",
                      },
                      "ref": null,
                      "rendered": "movie",
                      "type": "span",
                    },
                    "",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "2019",
                        "className": "hero__info__text__year",
                      },
                      "ref": null,
                      "rendered": "2019",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "hero__info__text__summary",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="hero__info__text__action-btns__favorite"
                    onClick={[Function]}
                  >
                    add to 
                     favorite
                  </button>,
                  <button
                    className="hero__info__text__action-btns__watch-later"
                    onClick={[Function]}
                  >
                    add to 
                     Watch Later
                  </button>,
                ],
                "className": "hero__info__text__action-btns",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "add to ",
                      " favorite",
                    ],
                    "className": "hero__info__text__action-btns__favorite",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "add to ",
                    " favorite",
                  ],
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "add to ",
                      " Watch Later",
                    ],
                    "className": "hero__info__text__action-btns__watch-later",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    "add to ",
                    " Watch Later",
                  ],
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="hero__info"
        >
          <Link
            onClick={[Function]}
            replace={false}
            to="/details/0"
          >
            <img
              alt=""
              className="hero__info__img"
              src="https://image.tmdb.org/t/p/w185/null"
            />
          </Link>
          <div
            className="hero__info__text"
          >
            <Link
              onClick={[Function]}
              replace={false}
              to="/details/0"
            >
              <h1
                className="hero__info__text__header"
              >
                some
              </h1>
              <div
                className="hero__info__text__rating"
              >
                4
                <RatingStars
                  stars={4}
                />
              </div>
              <div
                className="hero__info__text__type"
              >
                <span>
                  movie
                </span>
                
                <span
                  className="hero__info__text__year"
                >
                  2019
                </span>
              </div>
              <p
                className="hero__info__text__summary"
              />
            </Link>
            <div
              className="hero__info__text__action-btns"
            >
              <button
                className="hero__info__text__action-btns__favorite"
                onClick={[Function]}
              >
                add to 
                 favorite
              </button>
              <button
                className="hero__info__text__action-btns__watch-later"
                onClick={[Function]}
              >
                add to 
                 Watch Later
              </button>
            </div>
          </div>
        </div>,
        "className": "hero",
        "style": Object {
          "backgroundImage": "linear-gradient(to bottom, rgba(0,0,0,0.6), rgba(0,0,0,0.9)), url(https://image.tmdb.org/t/p/original/null)",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              onClick={[Function]}
              replace={false}
              to="/details/0"
            >
              <img
                alt=""
                className="hero__info__img"
                src="https://image.tmdb.org/t/p/w185/null"
              />
            </Link>,
            <div
              className="hero__info__text"
            >
              <Link
                onClick={[Function]}
                replace={false}
                to="/details/0"
              >
                <h1
                  className="hero__info__text__header"
                >
                  some
                </h1>
                <div
                  className="hero__info__text__rating"
                >
                  4
                  <RatingStars
                    stars={4}
                  />
                </div>
                <div
                  className="hero__info__text__type"
                >
                  <span>
                    movie
                  </span>
                  
                  <span
                    className="hero__info__text__year"
                  >
                    2019
                  </span>
                </div>
                <p
                  className="hero__info__text__summary"
                />
              </Link>
              <div
                className="hero__info__text__action-btns"
              >
                <button
                  className="hero__info__text__action-btns__favorite"
                  onClick={[Function]}
                >
                  add to 
                   favorite
                </button>
                <button
                  className="hero__info__text__action-btns__watch-later"
                  onClick={[Function]}
                >
                  add to 
                   Watch Later
                </button>
              </div>
            </div>,
          ],
          "className": "hero__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt=""
                className="hero__info__img"
                src="https://image.tmdb.org/t/p/w185/null"
              />,
              "onClick": [Function],
              "replace": false,
              "to": "/details/0",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "hero__info__img",
                "src": "https://image.tmdb.org/t/p/w185/null",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="/details/0"
                >
                  <h1
                    className="hero__info__text__header"
                  >
                    some
                  </h1>
                  <div
                    className="hero__info__text__rating"
                  >
                    4
                    <RatingStars
                      stars={4}
                    />
                  </div>
                  <div
                    className="hero__info__text__type"
                  >
                    <span>
                      movie
                    </span>
                    
                    <span
                      className="hero__info__text__year"
                    >
                      2019
                    </span>
                  </div>
                  <p
                    className="hero__info__text__summary"
                  />
                </Link>,
                <div
                  className="hero__info__text__action-btns"
                >
                  <button
                    className="hero__info__text__action-btns__favorite"
                    onClick={[Function]}
                  >
                    add to 
                     favorite
                  </button>
                  <button
                    className="hero__info__text__action-btns__watch-later"
                    onClick={[Function]}
                  >
                    add to 
                     Watch Later
                  </button>
                </div>,
              ],
              "className": "hero__info__text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <h1
                      className="hero__info__text__header"
                    >
                      some
                    </h1>,
                    <div
                      className="hero__info__text__rating"
                    >
                      4
                      <RatingStars
                        stars={4}
                      />
                    </div>,
                    <div
                      className="hero__info__text__type"
                    >
                      <span>
                        movie
                      </span>
                      
                      <span
                        className="hero__info__text__year"
                      >
                        2019
                      </span>
                    </div>,
                    <p
                      className="hero__info__text__summary"
                    />,
                  ],
                  "onClick": [Function],
                  "replace": false,
                  "to": "/details/0",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "some",
                      "className": "hero__info__text__header",
                    },
                    "ref": null,
                    "rendered": "some",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        4,
                        <RatingStars
                          stars={4}
                        />,
                      ],
                      "className": "hero__info__text__rating",
                    },
                    "ref": null,
                    "rendered": Array [
                      4,
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "stars": 4,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <span>
                          movie
                        </span>,
                        "",
                        <span
                          className="hero__info__text__year"
                        >
                          2019
                        </span>,
                      ],
                      "className": "hero__info__text__type",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "movie",
                        },
                        "ref": null,
                        "rendered": "movie",
                        "type": "span",
                      },
                      "",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "2019",
                          "className": "hero__info__text__year",
                        },
                        "ref": null,
                        "rendered": "2019",
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "hero__info__text__summary",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="hero__info__text__action-btns__favorite"
                      onClick={[Function]}
                    >
                      add to 
                       favorite
                    </button>,
                    <button
                      className="hero__info__text__action-btns__watch-later"
                      onClick={[Function]}
                    >
                      add to 
                       Watch Later
                    </button>,
                  ],
                  "className": "hero__info__text__action-btns",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "add to ",
                        " favorite",
                      ],
                      "className": "hero__info__text__action-btns__favorite",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "add to ",
                      " favorite",
                    ],
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "add to ",
                        " Watch Later",
                      ],
                      "className": "hero__info__text__action-btns__watch-later",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "add to ",
                      " Watch Later",
                    ],
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
